Network/AMQP/Internal.hs

```diff
@@ -486,6 +486,14 @@ data Channel = Channel {
                     chanExceptionHandlers :: MVar [CE.SomeException -> IO ()]
                 }
 
+data ChanThreadKilledException = ChanThreadKilledException { cause :: CE.SomeException }
```

Comment:
I'm wondering whether we should export that data-type? It might allow users to rethrow that exception in case they do some exception-handling the subscription-callback.